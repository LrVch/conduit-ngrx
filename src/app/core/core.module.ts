import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
