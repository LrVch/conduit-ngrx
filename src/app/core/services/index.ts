export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
