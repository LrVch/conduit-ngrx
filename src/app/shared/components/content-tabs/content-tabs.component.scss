.content-tabs {
    margin-bottom: 20px;
    min-height: 36px;
}