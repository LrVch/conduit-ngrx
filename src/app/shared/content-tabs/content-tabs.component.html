<div class="content-tabs">
  <button *ngFor="let tab of tabs"
      [color]="type === tab.value ? 'primary' : ''"
     mat-button
     (click)="onTabClick(tab.value, type)"
     >{{ tab.title }}</button>
  <ng-content></ng-content>
</div>