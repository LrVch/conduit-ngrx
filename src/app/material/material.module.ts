import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
];

@NgModule({
  imports: [...modules],
  exports: [...modules]
})
export class MaterialModule { }
