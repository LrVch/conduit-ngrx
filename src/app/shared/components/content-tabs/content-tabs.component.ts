import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

export interface Tab {
  title: string;
  value: string;
}

@Component({
  selector: 'app-content-tabs',
  templateUrl: './content-tabs.component.html',
  styleUrls: ['./content-tabs.component.scss']
})
export class ContentTabsComponent {
  @Input('type') type = '';
  @Input('tabs') tabs: Tab[] = [];
  @Output() selectedType = new EventEmitter<string>();

  onTabClick(tab: string, type: string): void {
    if (tab === type) {
      return;
    }

    this.selectedType.emit(tab);
  }
}
