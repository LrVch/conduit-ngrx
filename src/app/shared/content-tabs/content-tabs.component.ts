import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

interface Tab {
  title: string;
  value: string;
}

@Component({
  selector: 'app-content-tabs',
  templateUrl: './content-tabs.component.html',
  styleUrls: ['./content-tabs.component.scss']
})
export class ContentTabsComponent {
  @Input('type') type: string;
  @Output() selectedType = new EventEmitter<string>();
  @Input('tabs') tabs: Tab[];

  onTabClick(tab: string, type: string): void {
    if (tab === type) {
      return;
    }

    this.selectedType.emit(tab);
  }
}
