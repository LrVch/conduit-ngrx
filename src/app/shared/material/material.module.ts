import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
