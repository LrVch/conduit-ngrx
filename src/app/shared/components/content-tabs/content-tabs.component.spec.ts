import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '@app/lib/testing';

import { ContentTabsComponent } from './content-tabs.component';
import { NO_ERRORS_SCHEMA, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';

describe('ContentTabsComponent', () => {
  let component: ContentTabsComponent;
  let fixture: ComponentFixture<ContentTabsComponent>;
  let de: DebugElement;
  let el: HTMLElement;
  const event = {
    prevented: false,
    preventDefault() {
      this.prevented = true;
    }
  };

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [ContentTabsComponent],
        imports: [MaterialModule]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(ContentTabsComponent);
      component = fixture.componentInstance;
      de = fixture.debugElement;
      el = de.nativeElement;
      fixture.detectChanges();
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should has "type" @Input', () => {
    expect(component.type).toBe('');
  });

  it('should has "tabs" @Input', () => {
    expect(component.tabs).toEqual([]);
  });

  it('should show "tabs" @Input', () => {
    component.tabs = [
      { title: 'tab1', value: '1' },
      { title: 'tab2', value: '2' }
    ];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    component.tabs = [{ title: 'tab1', value: '1' }];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    component.tabs = [];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should hightlight selected tab with primary color', () => {
    component.tabs = [
      { title: 'tab1', value: '1' },
      { title: 'tab2', value: '2' }
    ];
    component.type = '1';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should raises selectedType event when clicked (onTabClick) and select first tab', () => {
    let selectedTab: string;
    const expectedTab = '1';
    component.tabs = [
      { title: 'tab1', value: '1' },
      { title: 'tab2', value: '2' }
    ];
    component.selectedType.subscribe((tab: string) => (selectedTab = tab));

    fixture.detectChanges();

    const firstTab = de.query(By.css('button'));

    firstTab.triggerEventHandler('click', null);
    expect(selectedTab).toBe(expectedTab);
  });

  it("should'n raises selectedType event when clicked (onTabClick) if selected tab is equals to passed type", () => {
    let wasEmited = false;

    component.type = '1';
    component.tabs = [
      { title: 'tab1', value: '1' },
      { title: 'tab2', value: '2' }
    ];
    component.selectedType.subscribe((tab: string) => {
      wasEmited = !!tab;
    });

    fixture.detectChanges();

    const firstTab = de.query(By.css('button'));

    firstTab.triggerEventHandler('click', null);
    expect(wasEmited).toBeFalsy();
  });
});
