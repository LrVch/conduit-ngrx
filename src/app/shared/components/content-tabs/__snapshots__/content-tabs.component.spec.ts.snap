// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTabsComponent should hightlight selected tab with primary color 1`] = `
<app-content-tabs
  selectedType={[Function EventEmitter]}
  tabs={[Function Array]}
  type={[Function String]}
>
  <div
    class="content-tabs"
  >
    
    <button
      class="mat-button mat-primary"
      mat-button=""
      ng-reflect-color="primary"
    >
      <span
        class="mat-button-wrapper"
      >
        tab1
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="mat-button"
      mat-button=""
      ng-reflect-color=""
    >
      <span
        class="mat-button-wrapper"
      >
        tab2
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</app-content-tabs>
`;

exports[`ContentTabsComponent should show "tabs" @Input 1`] = `
<app-content-tabs
  selectedType={[Function EventEmitter]}
  tabs={[Function Array]}
  type=""
>
  <div
    class="content-tabs"
  >
    
    <button
      class="mat-button"
      mat-button=""
      ng-reflect-color=""
    >
      <span
        class="mat-button-wrapper"
      >
        tab1
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="mat-button"
      mat-button=""
      ng-reflect-color=""
    >
      <span
        class="mat-button-wrapper"
      >
        tab2
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</app-content-tabs>
`;

exports[`ContentTabsComponent should show "tabs" @Input 2`] = `
<app-content-tabs
  selectedType={[Function EventEmitter]}
  tabs={[Function Array]}
  type=""
>
  <div
    class="content-tabs"
  >
    
    <button
      class="mat-button"
      mat-button=""
      ng-reflect-color=""
    >
      <span
        class="mat-button-wrapper"
      >
        tab1
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</app-content-tabs>
`;

exports[`ContentTabsComponent should show "tabs" @Input 3`] = `
<app-content-tabs
  selectedType={[Function EventEmitter]}
  tabs={[Function Array]}
  type=""
>
  <div
    class="content-tabs"
  >
    
  </div>
</app-content-tabs>
`;
